### PR TITLE
refactor: Update Langchain packages (no-changelog)

### PR DESCRIPTION
**Pull request by OlegIvaniv**
## Summary
Update Langchain packages to the latest versions and move shared `@langchain/core` to the pnpm-workspace catalog.

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/@n8n/nodes-langchain/package.json [hunk 1]
```diff
@@ -130,25 +130,26 @@
...
+    "@huggingface/inference": "2.8.0",
+    "@langchain/anthropic": "0.2.16",
+    "@langchain/cohere": "0.2.2",
+    "@langchain/community": "0.2.32",
+    "@langchain/core": "catalog:",
+    "@langchain/google-genai": "0.0.26",
+    "@langchain/google-vertexai": "0.0.27",
+    "@langchain/groq": "0.0.17",
+    "@langchain/mistralai": "0.0.29",
+    "@langchain/ollama": "^0.0.4",
```
Comment (burivuhster) [inline packages/@n8n/nodes-langchain/package.json:150]: Should we fix the exact version here?

File: packages/@n8n/nodes-langchain/package.json [hunk 2]
```diff
@@ -130,25 +130,26 @@
...
+    "@langchain/openai": "0.2.10",
+    "@langchain/pinecone": "0.0.9",
     "@langchain/qdrant": "^0.0.5",
```
Comment (burivuhster) [inline packages/@n8n/nodes-langchain/package.json:153]: And also here

**Review by burivuhster — COMMENTED** [review]
Please check the version pinning, otherwise good to go 👌

**Review by burivuhster — APPROVED** [review]
Thanks!

**Comment by Joffcom** [comment]
Got released with [`n8n@1.59.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.59.0)

**Pull request merged**